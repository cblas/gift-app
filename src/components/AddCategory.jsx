import {useState} from 'react'

export const AddCategory = ({onCategoryValue}) => {
  const [category, setCategory] = useState('');

  const handleInput = (e) => {
    setCategory(e.target.value);
  };

  const handleAdd = (e) => {
    e.preventDefault();

    //OPTION  1:
    //setCategories(categories => [...categories, category]);
    //OPTION2:
    onCategoryValue(category);

    setCategory('');
  };

  return (
    <form onSubmit={handleAdd}>
        <input 
            type="text" 
            onChange={handleInput}
            value={category}
            placeholder='Search by category...'
            />

    </form>
  )
}
