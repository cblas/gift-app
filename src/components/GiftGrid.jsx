import {GridItem} from './GridItem'
import { useGetImages } from '../hooks/useGetImages'


 export const GiftGrid = ({category}) => {
  //OPTION1:
  // const [images, setImages] = useState([]);

  // const getImages = async () => {
  //   const gifts = await getGiftsAsync(category);
  //   setImages(gifts);
  // }

  // useEffect(()=>{
  //   getImages();

  // },[category])
  
  //OPTION2: USE CUSTOM HOOK:
  const {images, isLoading} = useGetImages(category)

  return (
    <>
      <h3>{category}</h3>
      {isLoading && (<h4>Loading content...</h4>)}
      <div className='card-grid'>
            {
              images.map(image => {
                return <GridItem key={image.id} {...image} />
              })
               
            }
      </div>
                
    </>
  )
}
