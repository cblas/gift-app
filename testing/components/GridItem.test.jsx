import {render} from '@testing-library/react';
import {GridItem} from '../../src/components/GridItem';

describe('GridItem component testing ', () => {
    const title = 'my test';
    const url= 'https://i.blogs.es/82d7ef/pokemon/1366_2000.jpeg';

    test('shoul be do match with snapshot', () => {
        const {container} = render(<GridItem title={title} url ={url}/>)
        expect(container).toMatchSnapshot();
    })
});