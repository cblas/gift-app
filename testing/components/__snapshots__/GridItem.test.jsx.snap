// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem component testing  shoul be do match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="my test"
      src="https://i.blogs.es/82d7ef/pokemon/1366_2000.jpeg"
    />
    <p>
      my test
    </p>
  </div>
</div>
`;
